src/include/mpitimpl.h

```diff
@@ -1203,6 +1205,8 @@ extern MPID_Thread_mutex_t mpi_t_mutex;
         MPIR_T_THREAD_CHECK_BEGIN \
         MPID_Thread_mutex_destroy(&mpi_t_mutex, &err_); \
         MPIR_Assert(err_ == 0); \
+        MPID_Thread_finalize(&err_); \
+        MPIR_Assert(err_ == 0); \
```

Comment:
Same question, and this is trickier. We can't really finalize the thread until it is absolutely the last one. I guess we'll need keep a counter inside the MPL wrapper to ensure the balance.

Reply:
Thank you! There's an internal refcount in Argobots, so in the case of Argobots, it works fine.

Currently Pthreads and Windows threads do nothing in `MPL_thread_init()` and `MPL_thread_finalize()`, but since other threading layers might be added in the future, it is a good idea to keep a counter inside the MPL wrapper.

If this change is okay, I will add a new wrapper to keep a counter as you suggested.

Reply:
I see. For now, I would be satisfied by just adding a comment for `MPL_thread_init` (in each implementation) noting that they need balance init/finalize. As long as it is explicitly noted, we can add that counter in MPL when necessary -- it doesn't have to be now.

Reply:
Thanks. I will add a comment.